### DISCUSSION TITLE
How to evaluate capacity of Coturn server

### DISCUSSION THREAD

**Discussion by Tsukiand** [coturn/coturn#1217] [Q&A]
We have coturn cluster but we have no idea how much capacity it used. We just use network throughpu and udp packet loss rate as an indicator. We want to have a clear data to describe the capacity of coturn server.

**Comment by eakraly** [comment]
Hi @Tsukiand 

Capacity is a very elusive topic... It somewhat depends on the use case. If I understand correctly you want to know what is the upper limit for a specific machine.

I suggest following for you to try: overload a single instance of coturn with traffic as close to production as possible and see where it stops scaling with traffic.

This could be done with test traffic (for example generated by turnutils_uclient) or some other test infra if you have one or actual production.

What closely following metrics (not coturn exposed necessarily)
- CPU load
- Network bytes in/out
- Packets in/out

Once you see one of the flatten out (or your CPU getting to 100%) that is when you know you hit some limit. This is your "100%". From that knowledge you can figure out what is the limit and how you can use it.

**Reply by Tsukiand** [reply]
@eakraly Thanks for your replay, we have already done the performance test. We have the coturn for production usage, so we must guarantee the user experience. So the CPU/Network Throughout/Packets are not enough. We add udp data loss rate as indicator because big udp data loss rate cause bad user experience.

Another question: we want to have a data that describe the working status of coturn. (Whether it works well or not from coturn angle)

**Reply by eakraly** [reply]
In my past I used 2 signals for coturn health:
- TCP response for liveness check (if alive at all)
- Prometheus metric
